frontend/lib/item_listing/components/item_card/item_card.component.dart

```diff
@@ -17,38 +18,96 @@ class ItemCard extends StatelessWidget {
     this.category,
   });
 
+  Future<ImageProvider?> _fetchImage(String url) async {
+    final response = await dio.get(
+      url,
+      options: Options(responseType: ResponseType.bytes),
+    );
+    return MemoryImage(response.data);
+  }
+
   @override
   Widget build(BuildContext context) {
-    final imageWidgets = imageUrls != null && imageUrls!.isNotEmpty
-      ? Container(
+    Widget buildLoadingImage() {
```

Comment:
For readability sake, let's move the `buildXXX` helpers outside of the `build` method.